src/parser/parse_map.c

```diff
@@ -78,3 +57,34 @@ void	add_padding(t_game *game)
 	}
 	free(space_line);
 }
+
+static void	add_line_to_map(t_game *game, char *line)
+{
+	if (ft_strlen(line) > 0 && line[ft_strlen(line) - 1] == '\n')
+		line[ft_strlen(line) - 1] = '\0';
+	game->map = ft_realloc(game->map, sizeof(char *) * game->map_height,
+			sizeof(char *) * (game->map_height + 1));
+	if (game->map == NULL)
+	{
+		free(line);
+		exit_error_parser(game, game->map, ERR_MSG_FAILED_TO_REALLOC_MAP);
+	}
+	game->map[game->map_height] = ft_strdup(line);
+	if (game->map[game->map_height] == NULL)
+	{
+		free(line);
+		free_2d_array(game->map, game->map_height);
```

Comment:
if you got rid of new_map then free_2d_array is not necessary here, because it is inside the exit_error_parser